backend/hitas/views/condition_of_sale.py

```diff
@@ -0,0 +1,153 @@
+from hitas.utils import get_many_or_cached_prefetch, lookup_id_to_uuid
+from hitas.views.utils import HitasModelSerializer, HitasModelViewSet
+
+
+class MinimalApartmentSerializer(HitasModelSerializer):
+    class Meta:
+        model = Apartment
+        fields = [
+            "id",
+            "street_address",
+            "apartment_number",
+            "floor",
+            "stair",
+        ]
+
+
+class MinimalOwnerSerializer(HitasModelSerializer):
+    class Meta:
+        model = Owner
+        fields = [
+            "id",
+            "name",
+            "identifier",
+            "email",
+        ]
+
+
+class ConditionOfSaleOwnershipSerializer(HitasModelSerializer):
+
+    apartment = MinimalApartmentSerializer()
+    owner = MinimalOwnerSerializer()
+
+    class Meta:
+        model = Ownership
+        fields = [
+            "id",
+            "percentage",
+            "apartment",
+            "owner",
+        ]
+
+
+class ConditionOfSaleSerializer(EnumSupportSerializerMixin, HitasModelSerializer):
+
+    new_ownership = ConditionOfSaleOwnershipSerializer(read_only=True)
+    old_ownership = ConditionOfSaleOwnershipSerializer(read_only=True)
+    grace_period = EnumField(GracePeriod)
+    fulfilled = serializers.SerializerMethodField()
+
+    @staticmethod
+    def get_fulfilled(instance: ConditionOfSale) -> Optional[datetime.datetime]:
+        return instance.fulfilled
+
+    class Meta:
+        model = ConditionOfSale
+        fields = [
+            "id",
+            "new_ownership",
+            "old_ownership",
+            "grace_period",
+            "fulfilled",
+        ]
+
+
+class ConditionOfSaleCreateSerializer(serializers.Serializer):
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        owner_uuid = lookup_id_to_uuid(self.context["view"].kwargs.get("owner_uuid"), Owner)
+
+        try:
+            attrs["owner"] = Owner.objects.prefetch_related(
+                Prefetch(
+                    "ownerships",
+                    Ownership.objects.select_related("apartment").only(
+                        "id",
+                        "percentage",
+                        "apartment__id",
+                        "apartment__first_purchase_date",
+                        "apartment__latest_purchase_date",
+                        "apartment__completion_date",
+                    ),
+                ),
+                Prefetch(
+                    "ownerships__apartment__sales",
+                    ApartmentSale.objects.only(
+                        "id",
+                        "purchase_date",
+                    ).order_by("purchase_date"),
+                ),
+            ).get(uuid=owner_uuid)
+        except Owner.DoesNotExist as error:
+            raise HitasModelNotFound(model=Owner) from error
+
+        return attrs
+
+    def create(self, validated_data: dict[str, Any]) -> list[ConditionOfSale]:
```

Comment:
This method could use a comment about what/why is happening, as its a pretty irregular model endpoint

Reply:
Ok, added some explanation to this and the validate method.